labconnect/main/profile_routes.py

```diff
@@ -0,0 +1,104 @@
+            db.select(UserDepartments.department_id).where(
+                UserDepartments.user_id == user.id
+            )
+        )
+        .scalars()
+        .all()
+    )
+
+    user_majors = (
+        db.session.execute(
+            db.select(UserMajors.major_code).where(UserMajors.user_id == user.id)
+        )
+        .scalars()
+        .all()
+    )
+
+    return {
+        "id": user.id,
+        "email": user.email,
+        "first_name": user.first_name,
+        "last_name": user.last_name,
+        "preferred_name": user.preferred_name,
+        "class_year": user.class_year,
+        "profile_picture": user.profile_picture,
+        "website": user.website,
+        "description": user.description,
+        "departments": user_departments,
+        "majors": user_majors,
+    }
+
+
+@main_blueprint.route("/profile", methods=["GET"])
+@jwt_required()
+def get_profile() -> Response:
+    """GET /profile: current user profile"""
+    user_email = get_jwt_identity()
+    user = db.session.execute(
+        db.select(User).where(User.email == user_email)
+    ).scalar_one_or_none()
+
+    if not user:
+        return make_response(jsonify({"msg": "User not found"}), 404)
+
+    return jsonify(user_to_dict(user))
+
+
+@main_blueprint.route("/profile", methods=["PUT"])
+@jwt_required()
+def update_profile() -> Response:
+    """PUT /profile: Updates current user profile"""
+    user_email = get_jwt_identity()
+    user = db.session.execute(
+        db.select(User).where(User.email == user_email)
+    ).scalar_one_or_none()
+
+    if not user:
+        return make_response(jsonify({"msg": "User not found"}), 404)
+
+    json_data = request.get_json()
+    if not json_data:
+        return make_response(jsonify({"msg": "Missing JSON in request"}), 400)
+
+    # Update basic User fields
+    user.first_name = json_data.get("first_name", user.first_name)
+    user.last_name = json_data.get("last_name", user.last_name)
+    user.preferred_name = json_data.get("preferred_name", user.preferred_name)
+    user.class_year = json_data.get("class_year", user.class_year)
+    user.website = json_data.get("website", user.website)
+    user.description = json_data.get("description", user.description)
+
+    if "departments" in json_data:
+        db.session.query(UserDepartments).filter(
+            UserDepartments.user_id == user.id
+        ).delete()
+        for dept_id in json_data["departments"]:
+            if db.session.get(
+                RPIDepartments, dept_id
+            ):  # Ensure department exists before adding
+                new_user_dept = UserDepartments(user_id=user.id, department_id=dept_id)
+                db.session.add(new_user_dept)
+
+    if "majors" in json_data:
+        db.session.query(UserMajors).filter(UserMajors.user_id == user.id).delete()
+        for major_code in json_data["majors"]:
+            if db.session.get(Majors, major_code):  # Ensure major exists before adding
```

Comment:
db calls can be a bit costly and given the list of majors isn't super long, I think it may be useful to query once and check against the list instead of querying for each major